pkg/loop/internal/test/relayer.go

```diff
@@ -358,3 +361,145 @@ func RunRelayer(t *testing.T, relayer internal.Relayer) {
 		require.NoError(t, err)
 	})
 }
+
+func RunFuzzPluginRelayer(f *testing.F, relayerFunc func(*testing.T) internal.PluginRelayer) {
+	f.Add("ABC\xa8\x8c\xb3G\xfc", "", true, []byte{}, true, true, "")
+
+	f.Fuzz(func(
+		t *testing.T, fConfig string, fAccts string, fAcctErr bool,
+		fSigned []byte, fSignErr bool, fValsWErr bool, fErr string,
+	) {
+		keystore := fuzzerKeystore{
+			accounts:      []string{fAccts}, // fuzzer does not support []string type
+			acctErr:       fAcctErr,
+			signed:        fSigned,
+			signErr:       fSignErr,
+			valuesWithErr: fValsWErr,
+			errStr:        fErr,
+		}
+
+		ctx := tests.Context(t)
+		_, err := relayerFunc(t).NewRelayer(ctx, fConfig, keystore)
+
+		grpcUnavailableErr(t, err)
+	})
+}
+
+func RunFuzzRelayer(f *testing.F, relayerFunc func(*testing.T) internal.Relayer) {
+	f.Add([]byte{}, int32(-1), "ABC\xa8\x8c\xb3G\xfc", false, []byte{}, "", "", []byte{})
```

Comment:
It would probably make sense to include a few realistic sets of args here, like one relayer/plugin config pair for each provider type.

Reply:
Like multiple `f.Add` calls with those pairs? The approach I went with was inputs that might be more likely to cause problems.

Reply:
Agreed!

Reply:
The fuzz runtime will try to explore cases that expand the coverage of the code being fuzzed, so realistic test cases provides hints about how to get past validation and in to deeper parts of the stack early-on, without having to "discover" valid sets of inputs through randomization. We can still include problematic cases too.

Reply:
Ok. I just added an update to this. I think it is a little messy since it appears valid inputs are very common between all providers, but I'm not completely sure.

Reply:
They should each have particular differences in the RelayConfig or PluginConfig. There is also the chainreader/codec config to consider as well.

Reply:
Ok. All I have to go on right now is the values currently in test cases. Where can I see some valid inputs to some of these providers?

Reply:
We will probably have to look in the core tests for those. @nolag and @reductionista can help with chainreader/codec configs.

It is also possible that some products don't have any quirks. :shrug: